cmd/otel-allocator/collector/collector.go

```diff
@@ -77,15 +77,15 @@ func (k *Client) Watch(ctx context.Context, labelMap map[string]string, fn func(
 			if err != nil {
 				log.Printf("unable to create collector pod watcher")
 			}
-			if msg := runWatch(ctx, k, watcher.ResultChan(), collectorMap, fn); msg != "" {
+			if msg := runWatch(ctx, k, watcher.ResultChan(), collectorMap, fn, false); msg != "" {
 				log.Printf("Collector pod watch event stopped: %v", msg)
 				return
 			}
 		}
 	}()
 }
 
-func runWatch(ctx context.Context, k *Client, c <-chan watch.Event, collectorMap map[string]bool, fn func(collectors []string)) string {
+func runWatch(ctx context.Context, k *Client, c <-chan watch.Event, collectorMap map[string]bool, fn func(collectors []string), test bool) string {
```

Comment:
This is not really the way to go here... You might want to have a channel here, or perhaps accept a callback function.